numba/targets/linalg.py

```diff
@@ -2106,3 +2106,88 @@ def cond_impl(a, p=None):
         return impl(a, p)
 
     return cond_impl
+
+
+@overload(np.linalg.matrix_rank)
+def matrix_rank_impl(a, tol=None):
+    """
+    Computes rank for matrices and vectors.
+    The only issue that may arise is that because numpy uses double
+    precision lapack calls whereas numba uses type specific lapack
+    calls, some singular values may differ and therefore counting the
+    number of them above a tolerance may lead to different counts,
+    and therefore rank, in some cases.
+    """
+    ensure_lapack()
+
+    _check_linalg_1_or_2d_matrix(a, "matrix_rank")
+    sv_func = _compute_singular_values(a)
+
+    def _2d_matrix_rank_impl(a, tol):
+
+        @jit(nopython=True)
+        def get_rank(sv, t):
+            """
+            Gets rank from singular values with cut-off at a given tolerance
+            """
+            rank = 0
+            for k in range(len(sv)):
+                if sv[k] > t:
+                    rank = rank + 1
+                else:  # sv is ordered big->small so break on condition not met
+                    break
+            return rank
```

Comment:
Don't think so? In the suggested, if `sv=np.array([1.])` then `k` will be returned as `0`, when it ought be `1`?